javascript/swiggy.js

```diff
@@ -1,52 +1,108 @@
+        "price": 149,
+        "currency": "INR",
+        "vegetarian": true,
+        "imgUrl": "https://media.istockphoto.com/photos/chicken-biryani-directly-above-photo-picture-id1169141170?k=20&m=1169141170&s=612x612&w=0&h=EpCF3lQF2GBRaVApNELuE5xFQfv8fyQ_wWC52hmyxeo=",
+        "categories": ["biryani"]
+    }
+];
 
-const cartItems = {"lineItems":[{"id":"1121","name":"Plain Veg Biryani","quantity":2,"price":149,"currency":"INR"}],"shippingFee":0,"discount":0,"tax":0,"subTotal":149}
+const cartItems = {
+    "lineItems": [{
+        "id": "1121",
+        "name": "Plain Veg Biryani",
+        "quantity": 2,
+        "price": 149,
+        "currency": "INR"
+    }],
+    "shippingFee": 0,
+    "discount": 0,
+    "tax": 0,
+    "subTotal": 149
+}
 
 
 const mp = new Map();
-menuItems.forEach(element => {
-  element.categories.forEach(key => {
-    if(!mp.has(key)){
-      mp.set(key,[]);
+
+categories.forEach(element => {
+    if (!mp.has(element.displayName.toLowerCase())) {
+        mp.set(element.displayName.toLowerCase(), []);
     }
-    mp.get(key).push(element);
-  });
+});
+
+menuItems.forEach(element => {
+    element.categories.forEach(key => {
+        if (!mp.has(key)) {
+            mp.set(key, []);
+        }
+        mp.get(key).push(element);
+    });
 });
 
 const main = document.querySelector("main");
 
 const divCategories = document.querySelector(".categories");
 
-const  ulCategories = document.createElement("ul");
+const ulCategories = document.createElement("ul");
 categories.forEach(element => {
-  const liCategories = document.createElement("li");
-  liCategories.id = element.id;
-  liCategories.innerText = element.displayName;
-  ulCategories.append(liCategories); 
+    const liCategories = document.createElement("li");
+    liCategories.id = element.id;
+    liCategories.innerText = element.displayName;
+    ulCategories.append(liCategories);
 });
 divCategories.append(ulCategories);
 
 const divMenuItems = document.querySelector(".menu-items");
 
-for(const [key, value] of mp.entries()){
-  const divTop = document.createElement("div");
-  divTop.className = "top";
-  divTop.innerHTML = `<h3>${key}</h3>
+function addCategories(key) {
+    const divcontainer = document.createElement("div");
+    divcontainer.classList = "catContainer";
+    const divTop = document.createElement("div");
+    divTop.className = "top";
+    divTop.innerHTML = `<h3>${key}</h3>
                       <p>${mp.get(key).length} Items</p>`
-  divMenuItems.append(divTop);
+    divcontainer.append(divTop);
 
-  for(let i=0;i<mp.get(key).length;i++){
-    const divDiscription = document.createElement("div");
-    divDiscription.className = "itemDiscription";
-    divDiscription.innerHTML = `<div>
+    for (let i = 0; i < mp.get(key).length; i++) {
+        const divDiscription = document.createElement("div");
+        divDiscription.className = "itemDiscription";
+        divDiscription.innerHTML = `<div>
```

Comment:
Functions should be atomic and follow SRP. Keep the template creation logic in a separate function